python/lsst/meas/astrom/matchPessimisticB.py

```diff
@@ -0,0 +1,594 @@
+        numConsensus = self.config.numPatternConsensus
+        if ref_array.shape[0] < self.config.numBrightStars or \
+           src_array.shape[0] < self.config.numBrightStars:
+            numConsensus = 1
+
+        self.log.debug("Current tol maxDist: %.4f..." % maxMatchDistArcSec)
+        self.log.debug("Current shift: %.4f" % maxShiftArcseconds)
+        self.log.debug("Current rot: %.4f" % self.config.maxRotationDeg)
+
+        # Create our matcher object.
+        pyPPMb = PessimisticPatternMatcherB(ref_array[:, :3], self.log)
+
+        # Start the ineration over our tolerances.
+        for try_idx in xrange(self.config.matcherIterations):
+            if try_idx == 0:
+                # If we are on the first most stringent tolerance,
+                # the matcher should behave like an optimistic pattern
+                # matcher. Exiting at the first match.
+                matcher_struct = pyPPMb.match(
+                    source_catalog=src_array,
+                    n_check=self.config.numPointsForShapeAttempt,
+                    n_match=self.config.numPointsForShape,
+                    n_agree=1,
+                    max_n_patterns=self.config.numBrightStars,
+                    max_shift=maxShiftArcseconds,
+                    max_rotation=self.config.maxRotationDeg,
+                    max_dist=maxMatchDistArcSec,
+                    min_matches=minMatchedPairs,
+                    pattern_skip_array=np.array(
+                        match_tolerance.failedPatternList)
+                )
+            else:
+                # Once we fail and start softening tolerences we switch to
+                # pessimistic or consenus mode where 3 patterns must agree
+                # on a rotation before exiting. We double the match dist
+                # tolerance each round and add 1 to the pattern complexiy and
+                # two to the number of candidate spokes to check.
+                matcher_struct = pyPPMb.match(
+                    source_catalog=src_array,
+                    n_check=self.config.numPointsForShapeAttempt + 2 * try_idx,
+                    n_match=self.config.numPointsForShape + try_idx,
+                    n_agree=numConsensus,
+                    max_n_patterns=self.config.numBrightStars,
+                    max_shift=(self.config.maxOffsetPix *
+                               wcs.pixelScale().asArcseconds()),
+                    max_rotation=self.config.maxRotationDeg,
+                    max_dist=maxMatchDistArcSec * 2. ** try_idx,
+                    min_matches=minMatchedPairs,
+                    pattern_skip_array=np.array(
+                        match_tolerance.failedPatternList)
+                )
+            if matcher_struct is None and \
+               match_tolerance.lastPattern is None and \
+               try_idx == 0:
+                # If we found a pattern on a previous run and can't
+                # find an optimistic match with the harhest tolerances
```

Comment:
Corrected the typo and added the comma as suggested. You did read the comment correctly.